src/Faker/Extension/Container.php

```diff
@@ -39,7 +39,7 @@ public function __construct(array $definitions)
      *
      * @return Extension
      */
-    public function get($id)
+    public function get(string $id)
```

Comment:
don't add any pram tyoes. it'll break compatability with v1

Reply:
also, we'll have to drop support for php 7.1 if we want to do this :/

Reply:
php 7.2 is the first version to implement parameter type variance properly

Reply:
Correct but i am having doubts on the correct approach here though.

I do not mind dropping older php but the BC is something we have to look out for.

Another option is to accept these BC and push a 2.x in between but that would mean a lot of work there.

Reply:
My bad you're right. This will likely require a new major version as they've done here (eg) : https://github.com/thephpleague/container/releases/tag/4.0.0

Reply:
Actually, it doesn't require a new version. Like I said, we can drop support for PHP 7.2, and not change the param types, then we can safely support both version 1 and version 2.